src/applications/vaos/tests/e2e/community-care.cypress.spec.js

```diff
@@ -19,9 +20,13 @@ describe('VAOS community care flow', () => {
     );
     cy.injectAxe();
     // Select primary care
-    cy.get('input[value="323"]')
-      .focus()
-      .check();
+    cy.get('input[value="323"]', { timeout: Timeouts.normal })
+      .should('exist')
+      .then(checkbox => {
```

Comment:
What's different about using `.then()` here instead of chaining `.check()` after `.should`? My understanding is that Cypress will retry the previous get command until the should assertion passes.

Reply:
[Cypress documentation](https://docs.cypress.io/guides/core-concepts/introduction-to-cypress#Cypress-is-Not-Like-jQuery) explicitly prescribes the use of `.then()` in order to avoid flakiness and rely on Cypress' built-in retrying and timeouts.

Reply:
The way I read those docs is that the section you linked to is calling out `.then()` as a way to get direct access to a DOM node:

> In Cypress, when you want to interact with a DOM element directly, call .then() with a callback function that receives the element as its first argument.

But there's no need for direct DOM node access in the change here, it's just being re-wrapped and then other Cypress commands/assertions are run on it.

The docs about [how retries work](https://docs.cypress.io/guides/core-concepts/retry-ability) in Cypress seems more relevant:

> If the assertion that follows the cy.get() command fails, then the cy.get() command will requery the application's DOM again. Then Cypress will try the assertion against the elements yielded from cy.get(). If the assertion still fails, cy.get() will try requerying the DOM again, and so on until the cy.get() command timeout is reached.

Reply:
Fair points all around -- retry docs are certainly more relevant for explaining how Cypress' retry mechanism works!  I believe your understanding is correct as well. I'm trying to remember why we came up with this method, and I'm failing. I believe what may be happening here is that Cypress' built-in retry is not having the desired effect here because the vagaries of the UI and/or animations.

Perhaps the `.should('exist')` assertion is passing but the UI is still not ready for the `check` interaction. By wrapping that interaction in a `then` it retries the whole `then` block until it succeeds.